### DISCUSSION TITLE
How to setup retention time for telemtry log ?

### DISCUSSION THREAD

**Discussion by owensulei3841** [open-telemetry/opentelemetry-collector#9465] [Q&A] [answered]
Hi guys, just want to ask how to setup the retention time for for my collector telemtry log?
This is my collector information, but i don't know how to setup the retention time for the output_paths file.
I just want to make sure the file won't be too large and eat all my memory.
service:
  telemetry:
    logs:
      level: info
      output_paths: [ /abc/telemetry.log ]

Thank you

cc @antonjim-te

**Comment by antonjim-te** [comment]
Hi @owensulei3841 ,
There is not a way to set up a max size for the output file.
But you can use other strategies to limit to reduce the size of your output file:
- Reduce log level. For example, using `level: warn`
- Setting `sampling` policy. For example:
```
	 		sampling:
		   		enabled: true
		   		tick: 60s
		   		initial: 10
		   		thereafter: 0
```
Here there will be 10 messages per minute.
You can also ask the community to add a new configuration to limit the size of the output file and one the limit is reached it will start writing on the same file again. 

Thanks. I hope it was useful.

**Reply by owensulei3841** [reply]
Hi @antonjim-te , thank you so much for answer my question.
One more question for the sampling, I see the you set initial : 10 and tehreafter : 0, does it mean we will sampling 10% at the beginning, then no sample at all after that? Is there a doc to use the sampling? Thank you

**Reply by antonjim-te** [reply, marked as answer]
You can find here the description of the configuration fields https://github.com/open-telemetry/opentelemetry-collector/blob/f5a7315cf88e10c0bce0166b35d9227727deaa61/service/telemetry/config.go#L99

**Reply by owensulei3841** [reply]
Thanks a lot!
